### ISSUE TITLE
Initial scroll bar size is wrong in diff view and branch log view

### ISSUE THREAD

**Issue by stefanhaller** [jesseduffield/lazygit#2389]
When showing a longer diff, the scroll bar is initially too long; as you scroll down it gets shorter, until it eventually reaches its real size (or the minimum possible size). This makes it hard to quickly get a sense of how long a diff is.

The same happens in the branch log view.

I suppose this is because we initially only read part of the command output for efficiency, and only read more once you scroll down. Still, I wonder if it would be possible to read enough data initially so that the scroll bar reaches its minimum possible size (or eof is reached, whichever happens first).

![Recording 2023-01-24 at 14 06 01](https://user-images.githubusercontent.com/1225667/214302239-a2f38177-e408-4568-9d7a-46b2b406d5f7.gif)

**Comment by stefanhaller** [comment]
Ok, I digged a bit and found that it comes from this awesome commit, of course: 23bcc191802eb

I wonder if a fix could be as simple as something along these lines:
```diff
diff --git a/pkg/gui/tasks_adapter.go b/pkg/gui/tasks_adapter.go
index 33aa09eb..2bad208e 100644
--- a/pkg/gui/tasks_adapter.go
+++ b/pkg/gui/tasks_adapter.go
@@ -17,7 +17,6 @@ func (gui *Gui) newCmdTask(view *gocui.View, cmd *exec.Cmd, prefix string) error
 	).Debug("RunCommand")
 
 	_, height := view.Size()
-	_, oy := view.Origin()
 
 	manager := gui.getManager(view)
 
@@ -35,7 +34,13 @@ func (gui *Gui) newCmdTask(view *gocui.View, cmd *exec.Cmd, prefix string) error
 		return cmd, r
 	}
 
-	if err := manager.NewTask(manager.NewCmdTask(start, prefix, height+oy+10, nil), cmdStr); err != nil {
+	// We want to read as many lines initially as necessary to let the
+	// scrollbar go to its minimum height, so that the scrollbar thumb doesn't
+	// change size as you scroll down.
+	minScrollbarHeight := 2
+	linesToRead := height * (height - 1) / minScrollbarHeight
+
+	if err := manager.NewTask(manager.NewCmdTask(start, prefix, linesToRead, nil), cmdStr); err != nil {
 		gui.c.Log.Error(err)
 	}
 
```

It fixes the problem for me, and it doesn't make flicking through commits noticeably slower. (But I also have a fast machine... 😄) It could be a problem for people with huge monitors, as the number of lines we need to read initially grows with the window size.

**Comment by stefanhaller** [comment]
Having just switched to the awesome delta pager, I noticed that the same change has to go into `newPtyTask` (in `pty.go`).

Any thoughts about this approach? I have been running with this change for a week now, and I'm very happy with it.

**Comment by stefanhaller** [comment]
This bothers me enough that I decided to make [a PR](https://github.com/jesseduffield/lazygit/pull/2497). I did some non-scientific measurements (flipping through a few hundred long commits by holding down the up-arrow key with the fastest repeat rate that the OS lets me dial in), and I couldn't measure any difference even with a `linesToRead` value of 20000. Still, I decided to cap it at 5000 in the PR, just to be on the safe side.
